AsyncDisplayKit/Layout/ASDimension.h

```diff
@@ -12,56 +12,177 @@
+ */
+extern ASDimension ASDimensionMakeWithPoints(CGFloat points);
 
-extern ASRelativeDimension ASRelativeDimensionMake(ASRelativeDimensionType type, CGFloat value);
+/**
+ * Returns a dimension with the specified fraction value.
+ */
+extern ASDimension ASDimensionMakeWithFraction(CGFloat fraction);
 
-extern ASRelativeDimension ASRelativeDimensionMakeWithPoints(CGFloat points);
+/**
+ * Returns whether two dimensions are equal.
+ */
+extern BOOL ASDimensionEqualToDimension(ASDimension lhs, ASDimension rhs);
 
-extern ASRelativeDimension ASRelativeDimensionMakeWithFraction(CGFloat fraction);
+/**
+ * Returns a NSString representation of a dimension.
+ */
+extern NSString *NSStringFromASDimension(ASDimension dimension);
+
+/**
+ * Resolve this dimension to a parent size.
+ */
+extern CGFloat ASDimensionResolve(ASDimension dimension, CGFloat parent, CGFloat autoSize);
 
-extern ASRelativeDimension ASRelativeDimensionCopy(ASRelativeDimension aDimension);
+#define ASD(...) ASDimensionMake(__VA_ARGS__)
 
-extern BOOL ASRelativeDimensionEqualToRelativeDimension(ASRelativeDimension lhs, ASRelativeDimension rhs);
+@interface NSNumber (ASDimension)
+@property (nonatomic, readonly) ASDimension as_pointDimension;
+@property (nonatomic, readonly) ASDimension as_fractionDimension;
+@end
 
-extern NSString *NSStringFromASRelativeDimension(ASRelativeDimension dimension);
+#pragma mark - ASLayoutableSize
 
-extern CGFloat ASRelativeDimensionResolve(ASRelativeDimension dimension, CGFloat parent);
+/**
+ * Returns an ASLayoutableSize with default values.
+ */
+extern ASLayoutableSize ASLayoutableSizeMake();
+
+/**
+ * Returns an ASLayoutableSize with the specified CGSize values as width and height.
+ */
+extern ASLayoutableSize ASLayoutableSizeMakeFromCGSize(CGSize size);
+
+/**
+ * Returns whether two sizes are equal.
+ */
+extern BOOL ASLayoutableSizeEqualToLayoutableSize(ASLayoutableSize lhs, ASLayoutableSize rhs);
+
+/**
+ * Returns a string formatted to contain the data from an ASLayoutableSize.
+ */
+extern NSString *NSStringFromASLayoutableSize(ASLayoutableSize size);
+
+/**
+ * Resolve the given size to a parent size.
+ */
+extern ASSizeRange ASLayoutableSizeResolve(ASLayoutableSize size, const CGSize parentSize);
+
+/**
+ * Resolve the given size relative to a parent size and an auto size.
+ */
+extern ASSizeRange ASLayoutableSizeResolveAutoSize(ASLayoutableSize size, const CGSize parentSize, ASSizeRange autoASSizeRange);
 
 #pragma mark - ASSizeRange
 
-extern ASSizeRange ASSizeRangeMake(CGSize min, CGSize max);
+/**
+ * Creates an ASSizeRange with provided size as both min and max.
+ */
+ASOVERLOADABLE extern ASSizeRange ASSizeRangeMake(CGSize size);
 
-/** Creates an ASSizeRange with the provided size as both min and max */
-extern ASSizeRange ASSizeRangeMakeExactSize(CGSize size);
+/**
+ * Creates an ASSizeRange with provided min and max size.
+ */
+ASOVERLOADABLE extern ASSizeRange ASSizeRangeMake(CGSize min, CGSize max);
```

Comment:
Should we than not inline most (or all) methods from `ASDimension.h`, not only the `Make` functions?